kpi/paginators.py

```diff
@@ -1,7 +1,84 @@
 # -*- coding: utf-8 -*-
-from rest_framework import serializers, exceptions
-from rest_framework.pagination import LimitOffsetPagination, PageNumberPagination
+from django.core.paginator import InvalidPage
+from rest_framework import serializers
+from rest_framework.exceptions import NotFound
+from rest_framework.pagination import (
+    LimitOffsetPagination,
+    PageNumberPagination,
+    _positive_int
+)
 from rest_framework.reverse import reverse_lazy
+from rest_framework.utils.urls import remove_query_param, replace_query_param
+
+
+class DataPagination(PageNumberPagination):
+    """
+    Pagination class for submissions.
+    Because DataViewSet doesn't provide a real Django ORM QuerySet and pagination
+    is based on `start` and `limit` querystring parameters, some methods of
```

Comment:
If we had the `v2` endpoint use `offset` instead of `start`, could we use DRF's `LimitOffsetPagination`? And, would that free us from having to re-implement as many things as you've had to already, or would it not save any work?

Reply:
Awesome, thank you. I had to specify Limit values, so I refactored the `DataPagination` class.
it's way better like this.